Sourcery/Parsing/ParserComposer.swift

```diff
@@ -56,85 +56,101 @@ struct ParserComposer {
-                    tuple.elements.forEach({
-                        $0.type = resolveType($0.typeName, type, typealiases)
-                    })
-                }
+                resolveVariableTypes($0, of: type, resolve: resolveType)
+            }
+            type.methods.forEach {
+                resolveMethodTypes($0, of: type, resolve: resolveType)
             }
 
-            type.typealiases.forEach({ (_, alias) in
-                alias.type = resolveType(alias.typeName, type, typealiases)
-            })
+            if let enumeration = type as? Enum {
+                resolveEnumTypes(enumeration, types: unique, resolve: resolveType)
+            }
+        }
 
-            // find actual methods parameters types and their argument labels
-            for method in type.methods {
-                let argumentLabels: [String]
-                if let labels = method.selectorName.range(of: "(")
-                        .map({ method.selectorName.substring(from: $0.upperBound) })?
-                        .trimmingCharacters(in: CharacterSet(charactersIn: ")"))
-                        .components(separatedBy: ":")
-                        .dropLast() {
-                    argumentLabels = Array(labels)
-                } else {
-                    argumentLabels = []
-                }
+        let filteredTypes = unique.values.filter {
+            let accessLevel = AccessLevel(rawValue: $0.accessLevel)
+            let isPrivate = accessLevel == .private || accessLevel == .fileprivate
+            if isPrivate && self.verbose { print("Skipping \($0.kind) \($0.name) as it is private") }
+            return !isPrivate
+        }.sorted { $0.name < $1.name }
 
-                for (index, parameter) in method.parameters.enumerated() {
-                    if index < argumentLabels.count {
-                        parameter.argumentLabel = argumentLabels[index]
-                    }
+        updateTypeRelationships(types: filteredTypes)
+        return filteredTypes
+    }
 
-                    parameter.type = resolveType(parameter.typeName, type, typealiases)
-                }
+    typealias TypeResolver = (TypeName, Type?) -> Type?
 
-                if !method.returnTypeName.isVoid {
-                    method.returnType = resolveType(method.returnTypeName, type, typealiases)
+    private func resolveVariableTypes(_ variable: Variable, of type: Type, resolve: TypeResolver) {
+        variable.type = resolve(variable.typeName, type)
 
-                    if method.isInitializer {
-                        method.returnTypeName = TypeName("")
-                    }
-                }
+        variable.typeName.tuple?.elements.forEach({ tuple in
+            tuple.type = resolve(tuple.typeName, type)
+        })
+    }
+
+    private func resolveMethodTypes(_ method: Method, of type: Type, resolve: TypeResolver) {
+        let argumentLabels: [String]
+        if let labels = method.selectorName.range(of: "(")
```

Comment:
ah nevermind, they merged the pr I was waiting for so I can merge, just rebase on top of master